### DISCUSSION TITLE
Call external platform api with OAuth into autocomplete field

### DISCUSSION THREAD

**Discussion by svj30** [directus/directus#14681] [Archive]
Hi all,
I installed directus using docker. I created a registration form where one field fetches the course details from absorb platform(https://support.absorblms.com/hc/en-us/articles/222482188-Getting-Started-With-The-Absorb-RESTful-API-v1-). The absorb platform needs an OAUTH token to be sent. So I created another api that acts as a bridge between directus and absorb. The api which I  created works fine when I'm accessing it using postman. But after integration I'm getting **blocked:csp** issue in directus. Please find the screenshots below.

With the autocomplete api I'm trying to get the Name of  the courses in the field.

The code:

```js
 const express = require('express');
  const server = express();
  const app = express.Router();
  const bodyParser = require('body-parser');
  const axios = require('axios');
  const cors = require('cors');
  app.use(cors());
  app.use(bodyParser.urlencoded());
  app.use(bodyParser.json());
  app.get('/connect', (req,res)=>{
	  axios.post('https://[absorburl]/api/Rest/v1/Authenticate',{
                  "Username" : "[username]",
                  "Password" : "[password]",
                  "PrivateKey" : "[apikey]"
          }).then(function(auth){
		  console.log(auth.data)
                  const headers = {
    'Content-Type': 'application/json',
    'Authorization': auth.data
  }
  
  axios.get("https://[absorburl]/api/Rest/v1/Courses", {
      headers: headers
    })
    .then((response) => {
      console.log(response.data)
      res.send(response.data);
  
    })
  
          })
  })
  server.use(app);
  server.listen(3002, (path)=>{console.log(path) })
```

The response is below:
```json
[
  {
    "CourseType": "OnlineCourse",
    "Id": "...",
    "Name": "Docker",
    "Description": "<p><strong>Delivery Mode:</strong></p><p>Learn on your own schedule with self-paced online training and labs.</p>",
    "Notes": null,
    "ExternalId": null,
    "AccessDate": null,
    "ExpireType": 2,
    "ExpireDuration": {
      "Years": 0,
      "Months": 3,
      "Days": 0,
      "Hours": 0
    },
    "ExpiryDate": null,
    "ActiveStatus": 0,
    "TagIds": [
      "...",
      "...",
      "...",
      "..."
    ],
    "ResourceIds": [
      "..."
    ],
    "EditorIds": [
      "...",
      "..."
    ],
    "Prices": [],
    "CompetencyDefinitionIds": [],
    "PrerequisiteCourseIds": [],
    "PostEnrollmentCourseIds": [],
    "AllowCourseEvaluation": false,
    "CategoryId": null,
    "CertificateUrl": ...,
    "Audience": null,
    "Goals": null,
    "Vendor": null,
    "CompanyCost": null,
    "LearnerCost": null,
    "CompanyTime": null,
    "LearnerTime": null
  }
]
```

The configuration of autocomplete field is below:
<img width="859" alt="image" src="https://user-images.githubusercontent.com/110059338/181099259-dd7959d7-5e5e-4f42-b819-2caa6cf2f18f.png">

The error I'm facing is below:
<img width="1439" alt="image" src="https://user-images.githubusercontent.com/110059338/181105201-f60b726c-2800-4bd7-a513-c7e7c8e28c50.png">

**Comment by rijkvanzanten** [comment]
The default content security policy header rules in Directus prevent third-party URLs to be requested from the app. You can update these rules through the content security environment variables: https://docs.directus.io/self-hosted/config-options/#security

**Comment by svj30** [comment]
I installed directus using docker. So should I be updating the environmental variables in directus or the api app that i created? If it's in directus could you please help me how to do that because I'm confused about the process.

**Reply by rijkvanzanten** [reply]
Are you using a docker-compose.yml file, or just plain `docker run ..` command?

**Reply by svj30** [reply]
I do have a docker-compose file. I followed these steps that were mentioned.
https://docs.directus.io/self-hosted/installation/docker/

**Reply by rijkvanzanten** [reply]
In that case, add the environment variables for the content security policy headers in the docker-compose file 👍🏻

**Reply by svj30** [reply]
@rijkvanzanten I added the environment variables as per the screenshot attached. It is not working. Could you please help me?
<img width="885" alt="image" src="https://user-images.githubusercontent.com/110059338/182449884-2249fbd4-efcc-4cfc-a1db-ed9e2529a853.png">

**Reply by rijkvanzanten** [reply]
`https://*.https://directus.io` looks like an invalid value to me. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/Sources#sources for correct values for this header 🙂

**Reply by svj30** [reply]
@rijkvanzanten  I tried  "https://\*.directus.io"  it did not work. I placed "http://\*.localhost:8055" where the registration form is hosted it still gave the same error. Lastly i tried placing "http://\*localhost:3002" where the api is hosted, I still got the same error. Could you please help me with the correct url that I'm supposed to use in that place.

**Reply by rijkvanzanten** [reply]
The error in your browser indicates that it's trying to hit `http://172.18.0.4/connect`, try using that?

**Comment by svj30** [comment]
@rijkvanzanten  could you please help with this.

**Comment by svj30** [comment]
@rijkvanzanten I changed the instance from 172.18.0.4 to localhost:3002 
<img width="1292" alt="image" src="https://user-images.githubusercontent.com/110059338/183449149-21c3defb-86ac-4257-a01c-355787d88ccd.png">
<img width="1408" alt="image" src="https://user-images.githubusercontent.com/110059338/183449228-0a6e82dc-1e64-4feb-8be7-cf2eb3ad41f0.png">
 I tried giving "http://\*.localhost:3002" and it still did not work

**Reply by rijkvanzanten** [reply]
> I tried giving "http://*.localhost:3002" and it still did not work

That makes sense, as the URL is `http://localhost:3002`, not a subdomain of `localhost`.
